packages/@aws-cdk/aws-apprunner-alpha/lib/auto-scaling-configuration.ts

```diff
@@ -0,0 +1,196 @@
+   * The ARN of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+   * The Name of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationRevision: number;
+}
+
+/**
+ * The App Runner Auto Scaling Configuration.
+ *
+ * @resource AWS::AppRunner::AutoScalingConfiguration
+ */
+export class AutoScalingConfiguration extends cdk.Resource implements IAutoScalingConfiguration {
+  /**
+   * Imports an App Runner Auto Scaling Configuration from attributes
+   */
+  public static fromAutoScalingConfigurationAttributes(scope: Construct, id: string,
+    attrs: AutoScalingConfigurationAttributes): IAutoScalingConfiguration {
+    const autoScalingConfigurationName = attrs.autoScalingConfigurationName;
+    const autoScalingConfigurationRevision = attrs.autoScalingConfigurationRevision;
+
+    class Import extends cdk.Resource implements IAutoScalingConfiguration {
+      public readonly autoScalingConfigurationName = autoScalingConfigurationName;
+      public readonly autoScalingConfigurationRevision = autoScalingConfigurationRevision;
+      public readonly autoScalingConfigurationArn = cdk.Stack.of(this).formatArn({
+        resource: 'autoscalingconfiguration',
+        service: 'apprunner',
+        resourceName: `${attrs.autoScalingConfigurationName}/${attrs.autoScalingConfigurationRevision}`,
+      });
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Imports an App Runner Auto Scaling Configuration from its ARN
+   */
+  public static fromArn(scope: Construct, id: string, autoScalingConfigurationArn: string): IAutoScalingConfiguration {
+    const resourceParts = cdk.Fn.split('/', autoScalingConfigurationArn);
+
+    if (!resourceParts || resourceParts.length < 3) {
+      throw new Error(`Unexpected ARN format: ${autoScalingConfigurationArn}`);
+    }
+
+    const autoScalingConfigurationName = cdk.Fn.select(0, resourceParts);
+    const autoScalingConfigurationRevision = Number(cdk.Fn.select(1, resourceParts));
+
+    class Import extends cdk.Resource implements IAutoScalingConfiguration {
+      public readonly autoScalingConfigurationName = autoScalingConfigurationName;
+      public readonly autoScalingConfigurationRevision = autoScalingConfigurationRevision;
+      public readonly autoScalingConfigurationArn = autoScalingConfigurationArn;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+    * The ARN of the Auto Scaling Configuration.
+    * @attribute
+    */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+    * The name of the Auto Scaling Configuration.
+    * @attribute
+    */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationRevision: number;
```

Comment:
Thank you. Personally, I'm in favor of making the revision a string. 
However, I have some concerns.

The revision in the Cfn (L1 construct) is a number, but would it be appropriate to change it to a string in the L2 construct? I'm not sure if there's a possibility of it changing, so if you know anything about it, please let me know.

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apprunner.CfnAutoScalingConfiguration.html#attrautoscalingconfigurationrevision

Additionally, since the already implemented VpcConnector treats the revision as a number, I'm also concerned about having a different approach from that.

https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-apprunner-alpha.VpcConnector.html#vpcconnectorrevisionspan-classapi-icon-api-icon-experimental-titlethis-api-element-is-experimental-it-may-change-without-noticespan

Reply:
Thank you for the callout.

I checked the cfnspec and yes the type is Integer, which is number in TS.

```
"Attributes": {
        "AutoScalingConfigurationRevision": {
          "PrimitiveType": "Integer"
        },
```

I am fine having it as `number` for consistency here but I kind of feel if apprunner supports `latest` revision someday, which would be a breaking change to CFN then CDK would have breaking change as well.

@GavinZZ @paulhcsun  what do you think?

Reply:
Discussed with team internally. I think we don't need to over-optimize the type here. CFN schema type changes are not supposed to happen especially after it's already in CloudFormation resource schema registry.